src/module.cpp

```diff
@@ -536,8 +536,43 @@ void Module::AddGlobalVariable(const std::string &name, const Type *type, Expr *
                 // the same type as the global.  (But not if it's an
                 // ExprList; they don't have types per se / can't type
                 // convert themselves anyway.)
-                if (llvm::dyn_cast<ExprList>(initExpr) == nullptr)
+                if (llvm::dyn_cast<ExprList>(initExpr) == nullptr) {
                     initExpr = TypeConvertExpr(initExpr, type, "initializer");
+                } else {
+                    ExprList *exprList = llvm::dyn_cast<ExprList>(initExpr);
+
+                    // If we have exprList that initalizes const varying int, e.g.:
+                    // static const int x = { 0, 1, 2, 3 };
+                    // then we need to convert initExpr to constExpr that can
+                    // be utilized later in arithmetic expressions that can be
+                    // calculated in compile time (Optimize function call below),
+                    if (type->IsConstType() && type->IsVaryingType()) {
+                        const int N = g->target->getVectorWidth();
+                        int32_t *vals = (int32_t *)alloca(N * sizeof(int32_t));
+                        memset(vals, 0, N * sizeof(int32_t));
+                        int i = 0;
+                        bool canInitVaryingConst = true;
+                        for (const Expr *e : exprList->exprs) {
+                            const ConstExpr *ce = llvm::dyn_cast<ConstExpr>(e);
+                            if (!ce || ce->Count() != 1) {
+                                canInitVaryingConst = false;
+                                break;
+                            }
+                            const Type *t = ce->GetType();
+                            if (!t || !t->IsUniformType() || !t->IsAtomicType() ||
+                                ce->getBasicType() != AtomicType::TYPE_INT32) {
+
+                                canInitVaryingConst = false;
+                                break;
+                            }
+                            // TODO! support other types.
```

Comment:
It's very strange to have support only for one type. Why not to add others as well?

Reply:
Because I need only this type for coming patches.

Reply:
I understand that. But it is a user visible change. Someone may try to use it and it will be hard to explain why we support this for `int` type and not for other atomic types.
If you still want to proceed without adding support for other types, please at least create a Github issue (+ ideally lit test which will fail but it's up to you).

Reply:
OK, I have supported other basic types.